src/libraries/System.Threading.Tasks.Dataflow/tests/Dataflow/TransformManyBlockTests.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,663 @@
+                var target = generator();
+                DataflowTestHelpers.TestOfferMessage_AcceptsDataDirectly(target);
+                DataflowTestHelpers.TestOfferMessage_CompleteAndOffer(target);
+
+                target = generator();
+                await DataflowTestHelpers.TestOfferMessage_AcceptsViaLinking(target);
+                DataflowTestHelpers.TestOfferMessage_CompleteAndOffer(target);
+            }
+        }
+
+        [Fact]
+        public void TestPostAsyncEnumerable()
+        {
+            foreach (bool bounded in DataflowTestHelpers.BooleanValues)
+            {
+                var tb = new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable, new ExecutionDataflowBlockOptions { BoundedCapacity = bounded ? 1 : -1 });
+                Assert.True(tb.Post(0));
+                tb.Complete();
+                Assert.False(tb.Post(0));
+            }
+        }
+
+        [Fact]
+        public Task TestCompletionTaskAsyncEnumerable()
+        {
+            return DataflowTestHelpers.TestCompletionTask(() => new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable));
+        }
+
+        [Fact]
+        public async Task TestLinkToOptionsAsyncEnumerable()
+        {
+            const int Messages = 1;
+            foreach (bool append in DataflowTestHelpers.BooleanValues)
+            {
+                var tb = new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable);
+                var values = new int[Messages];
+                var targets = new ActionBlock<int>[Messages];
+                for (int i = 0; i < Messages; i++)
+                {
+                    int slot = i;
+                    targets[i] = new ActionBlock<int>(item => values[slot] = item);
+                    tb.LinkTo(targets[i], new DataflowLinkOptions { MaxMessages = 1, Append = append });
+                }
+
+                tb.PostRange(0, Messages);
+                tb.Complete();
+                await tb.Completion;
+
+                for (int i = 0; i < Messages; i++)
+                {
+                    Assert.Equal(
+                        expected: append ? i : Messages - i - 1,
+                        actual: values[i]);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task TestReceivesAsyncEnumerable()
+        {
+            for (int test = 0; test < 2; test++)
+            {
+                var tb = new TransformManyBlock<int, int>(i => AsyncEnumerable.Repeat(i * 2, 1));
+                tb.PostRange(0, 5);
+
+                for (int i = 0; i < 5; i++)
+                {
+                    Assert.Equal(expected: i * 2, actual: await tb.ReceiveAsync());
+                }
+
+                Assert.False(tb.TryReceive(out _));
+                Assert.False(tb.TryReceiveAll(out _));
+            }
+        }
+
+        [Fact]
+        public async Task TestCircularLinkingAsyncEnumerable()
+        {
+            const int Iters = 200;
```

Comment:
```suggestion
            const int iters = 200;
```
private constants should be written in lower case